### DISCUSSION TITLE
Chain action getting canceled by parent task

### DISCUSSION THREAD

**Discussion by grinder81** [pointfreeco/swift-composable-architecture#2471] [General]
Hey folks, 

I was having some issues when a parent action was canceled, the long-chain effect is getting canceled as well. To explain the issue I used the TCA `VoiceMemos`  example app. To reproduce the issue, here are the steps:
1. Go to `VoiceMemos` reducer 
2. Around line 70 change the code as follows: 
```      
case let .recordingMemo(.presented(.delegate(.didFinish(.success(recordingMemo))))):
        state.recordingMemo = nil
        state.voiceMemos.insert(
          VoiceMemo.State(
            date: recordingMemo.date,
            duration: recordingMemo.duration,
            url: recordingMemo.url
          ),
          at: 0
        )
          return .run { send in
              try await self.clock.sleep(for: .seconds(5))
              print("Sleeping is over")
          }
```
3. The main change is 
```
return .run { send in
              try await self.clock.sleep(for: .seconds(5))
              print("Sleeping is over")
          }
```
instead of `return .none`
4. Of course you need to include the `clock` dependency 

From this code what do we expect? 
1. We start recording 
2. Stop recording after a while 
3. `state.voiceMemos` is saving the new recording 
4. And finally we are executing another task where we are sleeping for 5 sec 
5. After 5 second we should expect the print statment 

But that doesn't happen in that case.

After digging more and more I found following code is the culprit:
```
        async let startRecording: Void = send(
          .audioRecorderDidFinish(
            TaskResult { try await self.audioRecorder.startRecording(url) }
          )
        )
```
Basically when recording is ending, somehow it's canceling my sleeping task but why is that? If I change that code instead of the recording then everything is just working fine and I'm getting the sleeping print nicely. Please correct me if I'm thinking something wrong here.

**Comment by grinder81** [comment]
To identify or dig more, I would love to know when exactly that effects is getting canceled. I think that would be very helpful.

**Reply by Sajjon** [reply]
@grinder81 are you using any `.task` ViewModifier in the parent or child features view?
